apps/src/dance/ai/DanceAiClient.ts

```diff
@@ -14,131 +14,112 @@ export type CachedWeightsMapping = {
  * the score for each returned value.
- * @param {array} emojis: the list of emojis the user selected
+ * @param {array} selectedEmojis: the list of three emojis the user selected
  * @param {ChooseEffectsQuality} quality: whether we want good or bad scoring effects
- * @returns: an object containing the effects that were chosen, and their scores, for example:
+ * @returns {GeneratedEffect} an object containing the effects that were chosen, for example:
  * {
- *   "results: {
- *     "backgroundEffect": "sparkles",
- *     "backgroundColor": "cool",
- *     "foregroundEffect": "bubbles"
- *   },
- *   "scores": {
- *     "backgroundEffect": 0.1,
- *     "backgroundColor":  0.2,
- *     "foregroundEffect": 0.3
- *   }
+ *   "backgroundEffect": "sparkles",
+ *   "backgroundColor": "cool",
+ *   "foregroundEffect": "bubbles"
  * }
  */
 export function chooseEffects(
-  emojis: string[],
+  selectedEmojis: string[],
   quality: ChooseEffectsQuality
 ): GeneratedEffect {
-  // Obtain final summed output weight based off input received
-  const outputTypes: CachedWeightsMapping[] = [
-    CachedBackgrounds,
-    CachedForegrounds,
-    CachedPalettes,
-  ];
-  const outputWeights: number[][] = outputTypes.map(function (map) {
-    return calculateOutputWeightsVector(emojis, map);
-  });
-
-  // Sort and slice top scoring options, mapped to their output identifiers (e.g. [[0.25, 'squiggles'], ...])
-  const numRandomOptions = 3;
-  const outputOptions: [number, string][][] = outputWeights.map(
-    (weightVector, i) => {
-      const optionsAll = obtainOptions(weightVector, outputTypes[i]);
-      const options =
-        quality === ChooseEffectsQuality.GOOD
-          ? optionsAll.slice(0, numRandomOptions)
-          : optionsAll.slice(-numRandomOptions);
-      return options;
-    }
-  );
-
-  // Clarification of which array correlates to which option.
-  const backgroundOptions: [number, string][] = outputOptions[0];
-  const foregroundOptions: [number, string][] = outputOptions[1];
-  const paletteOptions: [number, string][] = outputOptions[2];
-
-  // Choose random values.
-  const backgroundEffectOption =
-    backgroundOptions[Math.floor(Math.random() * backgroundOptions.length)];
-  const backgroundColorOption =
-    paletteOptions[Math.floor(Math.random() * paletteOptions.length)];
-  const foregroundEffectOption =
-    foregroundOptions[Math.floor(Math.random() * foregroundOptions.length)];
-
-  const chosenEffects = {
-    backgroundEffect: backgroundEffectOption[1],
-    backgroundColor: backgroundColorOption[1],
-    foregroundEffect: foregroundEffectOption[1],
+  const chosenEffects: GeneratedEffect = {
+    [FieldKey.BACKGROUND_EFFECT]: '',
+    [FieldKey.FOREGROUND_EFFECT]: '',
+    [FieldKey.BACKGROUND_PALETTE]: '',
   };
-
+  for (const field of [
+    FieldKey.BACKGROUND_EFFECT,
+    FieldKey.FOREGROUND_EFFECT,
+    FieldKey.BACKGROUND_PALETTE,
+  ]) {
```

Comment:
Oh yeah I guess this could also just be `for (const field of Object.keys(chosenEffects)` since that's already been declared

Reply:
Changed to `for (const field of Object.keys(chosenEffects) as FieldKey[]) {`
thanks!

Reply:
Ahh interesting that you need to cast with `Object.keys` - as an alternative, I think you could also use `for (const field of Object.values(FieldKey)` and I think that shouldn't require a cast (just tried manually). If you prefer, you could also use that below in the helper function that iterates through the same keys.

Reply:
Nice! Just pushed change.